pkg/dispatcher/dispatcher.go

```diff
@@ -1,46 +1,95 @@
-package catalog
+package dispatcher
```

Comment:
We should discuss this because this has some coupling implications.
to avoid lengthy writes i'll reach offline.